File: core/src/main/java/com/exodus/game/ExodusGame.java
package com.exodus.game;

import com.badlogic.gdx.Game;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.GL30;
import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.utils.viewport.FitViewport;
import com.badlogic.gdx.utils.viewport.Viewport;

public class ExodusGame extends Game {
    private SpriteBatch batch;
    private OrthographicCamera camera;
    private Viewport viewport;
    
    // Virtual dimensions for consistent rendering across different screen sizes
    public static final float VIRTUAL_WIDTH = 1280;
    public static final float VIRTUAL_HEIGHT = 720;

    @Override
    public void create() {
        batch = new SpriteBatch();
        
        // Setup camera and viewport
        camera = new OrthographicCamera();
        viewport = new FitViewport(VIRTUAL_WIDTH, VIRTUAL_HEIGHT, camera);
        camera.position.set(VIRTUAL_WIDTH / 2, VIRTUAL_HEIGHT / 2, 0);
        
        // TODO: Initialize game screens and assets
    }

    @Override
    public void render() {
        // Update camera
        camera.update();
        batch.setProjectionMatrix(camera.combined);
        
        // Clear screen
        Gdx.gl.glClearColor(0.1f, 0.1f, 0.1f, 1);
        Gdx.gl.glClear(GL30.GL_COLOR_BUFFER_BIT);
        
        // Render game state
        super.render();
    }

    @Override
    public void resize(int width, int height) {
        viewport.update(width, height, true);
    }

    @Override
    public void dispose() {
        batch.dispose();
    }

    public SpriteBatch getBatch() {
        return batch;
    }

    public OrthographicCamera getCamera() {
        return camera;
    }

    public Viewport getViewport() {
        return viewport;
    }
}

File: core/src/main/kotlin/biz/dvlop/exodus/ExodusGame.kt
package biz.dvlop.exodus

import biz.dvlop.exodus.assets.GameAssets
import biz.dvlop.exodus.screens.GameScreen
import biz.dvlop.exodus.screens.MainGameScreen
import biz.dvlop.exodus.screens.MainMenuScreen
import com.badlogic.gdx.Game
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.g2d.SpriteBatch

/**
 * Main game class for Exodus.
 *
 * This class serves as the core game engine, handling the SpriteBatch,
 * font resources, and screen management.
 */
class ExodusGame : Game() {
    // Game assets and resources
    lateinit var batch: SpriteBatch
        private set
    lateinit var assets: GameAssets
        private set

    // Game screens
    lateinit var menuScreen: MainMenuScreen
        private set
    lateinit var gameScreen: MainGameScreen
        private set

    override fun create() {
        // Log the device information for debugging
        Gdx.app.log("ExodusGame", "Device: ${Gdx.app.type}, " +
                "Resolution: ${Gdx.graphics.width}x${Gdx.graphics.height}, " +
                "Density: ${Gdx.graphics.density}")

        // Initialize game resources
        batch = SpriteBatch()
        assets = GameAssets()
        assets.load()


        // Initialize screens
        menuScreen = MainMenuScreen(this)
        gameScreen = MainGameScreen(this)

        // Set initial screen
        setScreen(menuScreen)
    }

    override fun render() {
        super.render()
    }

    override fun dispose() {
        batch.dispose()
        assets.dispose()
        screen?.dispose()
    }
}
